nxdrive/engine/dao/sqlite.py

```diff
@@ -2735,6 +2735,24 @@ def resume_transfer(
             else:
                 self.transferUpdated.emit()
 
+    def resume_session(self, uid: int) -> None:
+        """Resume all transfers for given session."""
+        with self.lock:
+            c = self._get_write_connection().cursor()
+            c.execute(
+                "UPDATE Uploads SET status = ? WHERE doc_pair IN (SELECT id FROM States WHERE Session = ?)",
+                (TransferStatus.ONGOING.value, uid),
+            )
+            rows = c.execute(
+                "SELECT * FROM States WHERE id IN"
+                " (SELECT doc_pair FROM Uploads WHERE doc_pair IN (SELECT id FROM States WHERE Session = ?))",
+                (uid,),
+            ).fetchall()
+
+            for doc_pair in rows:
+                self.queue_manager.push(doc_pair)
+                self.directTransferUpdated.emit()
```

Comment:
To be even more efficient, should that line be unindented? It would trigger the signal only once, after all pairs are in the queue.